src/main/java/org/opensearch/knn/index/codec/util/KNNVectorAsArraySerializer.java

```diff
@@ -31,8 +33,8 @@ public byte[] floatToByteArray(float[] input) {
     }
 
     @Override
-    public float[] byteToFloatArray(ByteArrayInputStream byteStream) {
-        try {
+    public float[] byteToFloatArray(BytesRef bytesRef) {
+        try (ByteArrayInputStream byteStream = new ByteArrayInputStream(bytesRef.bytes, bytesRef.offset, bytesRef.length)) {
```

Comment:
what's the meaning of `bytesRef.offset`, are there any edge case scenarios where behavior is different from ByteArrayInputStream?

Reply:
adding to previous comment - should we add more checks if with offset we get out of the stream's boundaries? I mean in the code itself and also add a unit test that models such scenario.

Reply:
Class is defined here: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/BytesRef.java. 

> are there any edge case scenarios where behavior is different from ByteArrayInputStream

I dont think so - ByteArrayInputStream is pretty lightweight.

The constructor does do a validity check on BytesRef, so Im not sure we need to also check here for validity of bytes ref: https://github.com/apache/lucene/blob/5e52b8094aca4dab2898c7dd17084fc7cf7fd016/lucene/core/src/java/org/apache/lucene/util/BytesRef.java#L58. 

That being said, I think there is a mistake for this here: https://github.com/opensearch-project/k-NN/blob/50539182580aff1b4a26b684da39841bb78cfd1b/src/main/java/org/opensearch/knn/index/codec/util/KNNVectorAsCollectionOfFloatsSerializer.java#L31-L34. length already factors in offset so I think this is wrong. It has not showed up yet because offset is always 0 and length=length (i believe). 

Anyway, I will fix and add a unit test.